appium/webdriver/extensions/gsm.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from selenium import webdriver
+from ..mobilecommand import MobileCommand as Command
+
+
+class Gsm(webdriver.Remote):
+
+    (
+        NONE_OR_UNKNOWN,
+        POOR,
+        MODERATE,
+        GOOD,
+        GREAT
+    ) = list(map(str, range(5)))  # TODO: Need to check where int 0 will be dropped as arg
+
+    def set_gsm_signal(self, strength):
+        """Set GSM signal strength (Emulator only)
+
+        :Args:
+         - strength: Signal strength. Can be set Gsm.NONE_OR_UNKNOWN/POOR/MODERATE/GOOD/GREAT
+
+        :Usage:
+            self.driver.set_gsm_signal(Gsm.GOOD)
+        """
+        self.execute(Command.SET_GSM_SIGNAL, {'signalStrength': strength})
```

Comment:
Which Appium version did you run on your local?
We should send two params here because of https://github.com/appium/appium-base-driver/pull/308

```
{'signalStrength': strength,'signalStrengh': strength})
```

Reply:
> Which Appium version did you run on your local?

v1.12.1

Thanks for your notice, I'll support both params.

Reply:
@KazuCocoa 

> We should send two params here because of appium/appium-base-driver#308

Let me confirm, this reason is to cover old appium case, right?
(If so, I'll check another client behavior like java.)

* appium v1.10 (accept only "signalStrengh")

I'm having below error log when sending ```{'signalStrength': strength,'signalStrengh': strength})```

```
➜  mobile git:(master) ✗ python SampleAndroidTest.py
test_01_test_name (__main__.SimpleTest) ... 0
ERROR

======================================================================
ERROR: test_01_test_name (__main__.SimpleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "SampleAndroidTest.py", line 42, in test_01_test_name
    self.driver.set_gsm_signal(Gsm.POOR)
  File "/Users/atsushimori/.pyenv/versions/3.7.1/lib/python3.7/site-packages/appium/webdriver/extensions/gsm.py", line 38, in set_gsm_signal
    self.execute(Command.SET_GSM_SIGNAL, {'signalStrength': strength, 'signalStrengh': strength})
  File "/Users/atsushimori/.pyenv/versions/3.7.1/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/Users/atsushimori/.pyenv/versions/3.7.1/lib/python3.7/site-packages/appium/webdriver/errorhandler.py", line 29, in check_response
    raise wde
  File "/Users/atsushimori/.pyenv/versions/3.7.1/lib/python3.7/site-packages/appium/webdriver/errorhandler.py", line 24, in check_response
    super(MobileErrorHandler, self).check_response(response)
  File "/Users/atsushimori/.pyenv/versions/3.7.1/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.InvalidArgumentException: Message: Parameters were incorrect. We wanted {"required":["signalStrengh"]} and you sent ["signalStrength","signalStrengh","sessionId"]


----------------------------------------------------------------------
Ran 1 test in 22.366s

FAILED (errors=1)
```

Reply:
ah, yeah. The current way looks good.
Thanks for it.

Reply:
Changed to ```{'signalStrength': strength,'signalStrengh': strength})``` for https://github.com/appium/python-client/pull/357#discussion_r279664482